File: week17/week17-4.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        N=len(nums)
        big,small=False,False
        for i in range(N-1):
            d=nums[i+1]-nums[i]
            if d>0: big=True
            if d<0: small=True
        if big and small:return False
        else: return True

File: week17/week17-4.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int N= nums.size();
        int big=0,small=0;
        for(int i=0;i<N-1;i++){
            int d= nums[i+1] - nums[i];
            if(d>0) big=1;
            if(d<0) small=1;
        }
        if(big==1 && small==1) return false;
        else return true;
    }
};